File: golang/1927.sum-game.go
func sumGame(num string) bool {
    collect := func(l, r int) (n, q int) {
      for k := l; k < r; k++ {
        if num[k] == '?' {
          q++
        } else {
          n += int(num[k] - '0')
        }
      }

      return n, q
    }

    n := len(num)
    n0, q0 := collect(0, n / 2)
    n1, q1 := collect(n / 2, n)
    
    return (q0 + q1) % 2 == 1 || n0 - n1 != (q1 - q0) * 9 / 2
}

File: rust/1927.sum-game.rs
impl Solution {
    pub fn sum_game(num: String) -> bool {
        let num = num.as_bytes();
        
        let collect = |l: usize, r: usize| -> (i32, i32) {
            let mut n = 0;
            let mut q = 0;

            for k in (l..r) {
              if num[k] == b'?' {
                q += 1;
              } else {
                n += (num[k] - b'0') as i32;
              }
            }

            (n, q)
        };

        let n = num.len();
        let (n0, q0) = collect(0, n / 2);
        let (n1, q1) = collect(n / 2, n);

        (q0 + q1) % 2 == 1 || n0 - n1 != (q1 - q0) * 9 / 2
    }
}